### PR TITLE
[BEAM-4130] Add tests for FlinkJobServerDriver

### PR DESCRIPTION
**Pull request by mxm**
This adds a few test cases for FlinkJobServerDriver. It also changes the default
host from empty host to `localhost`.

Backported from master (cfaa01ad5eaf9ea7537646b2ca6cc1d8064ed872)

CC @tweise @angoenka 

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

File: runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkJobServerDriver.java [hunk 1]
```diff
@@ -54,44 +55,44 @@
...
 
     @Option(
       name = "--clean-artifacts-per-job",
       usage = "When true, remove each job's staged artifacts when it completes"
     )
-    private Boolean cleanArtifactsPerJob = false;
+    boolean cleanArtifactsPerJob = false;
 
     @Option(name = "--flink-master-url", usage = "Flink master url to submit job.")
-    private String flinkMasterUrl = "[auto]";
```
[the remark this answers is not in the document]
Comment (mxm) [reply runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkJobServerDriver.java]: Testing. They are package-local now.

**Comment by mxm** [comment]
@tweise Should be good to go now.

File: runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkJobServerDriver.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (tweise) [reply runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkJobServerDriver.java]: Few lines further :) There is no need to have a getter, especially not after the visibility of that was changed, too. Somehow it feels as couple recent changes in this area could have benefitted from more thorough review? (BTW I don't expect this to be addressed as part of this PR, but in master.)

[the remark this answers is not in the document]
Comment (mxm) [reply runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkJobServerDriver.java]: Considering that the code was more or less untested, I think we are much better off.

We can change the getter back to public.

[the remark this answers is not in the document]
Comment (mxm) [reply runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkJobServerDriver.java]: This is really a minor issue considering that FlinkJobServerDriver is not user-facing at all.

[the remark this answers is not in the document]
Comment (tweise) [reply runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkJobServerDriver.java]: I would probably remove it, but as it stands it is redundant.

The host default change that broke pipeline submission and this PR reverts was user facing. And yes, test coverage added here puts us into a much better place. Test coverage is always preferred, but looking at the PRs a bit more closely sometimes also helps.

[the remark this answers is not in the document]
Comment (mxm) [reply runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkJobServerDriver.java]: Yes, we should be careful. That said, all PRs were reviewed and mistakes do happen.

It helps to cut off the release branch a bit earlier so there is some time for testing.

**Comment by tweise** [comment]
Run Java PreCommit

**Comment by tweise** [comment]
@aaltay not that we are waiting to merge this PR - it is blocked by unrelated Java pre-commit issues.

**Comment by tweise** [comment]
Run Java PreCommit

**Comment by tweise** [comment]
Run Java PreCommit

**Pull request merged**